### PR TITLE
[Torch] Workaroud for torch.quantile function

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

torch quantile is always using numpy realization

### Reason for changes

Bug https://github.com/pytorch/pytorch/issues/64947

third party test 237 has failed

```
       RuntimeError: quantile() input tensor is too large
 ```
### Tests
Build test third party 239 is finished successfully

### CODE REVIEW

File: nncf/torch/tensor_statistics/collectors.py [hunk 1]
```diff
@@ -173,17 +173,10 @@ def quantile(
...
-            result = torch.quantile(
-                tensor.tensor,
-                torch.tensor(quantile, dtype=tensor.tensor.dtype, device=tensor.tensor.device),
-                axis,
-                keepdims,
-            )
+        # https://github.com/pytorch/pytorch/issues/64947
+        result = torch.tensor(
+            np.quantile(tensor.tensor.detach().cpu().numpy(), q=quantile, axis=axis, keepdims=keepdims)
+        )
```
Comment (vshampor) [inline nncf/torch/tensor_statistics/collectors.py:179]: If the shortcomings of the torch implementation are limited to the CPU version, then the code should only fall back to the numpy implementation if the original tensor device is CPU.

Comment (daniil-lyakhov) [reply nncf/torch/tensor_statistics/collectors.py]: I'm now using `torch.quantize` in case it is possible

**Comment by daniil-lyakhov** [comment]
@vshampor, please merge if you don't have any comments

**Comment by limaodaxia** [comment]
> @vshampor, please merge if you don't have any comments

good job

**Pull request merged**